uftrace info command fails

I installed uftrace and ran "uftrace info" command as followings.
Could you please why "uftrace info" command failed?

```
gohkim@ws00837:~/work/tools$ git clone https://github.com/namhyung/uftrace.git
Cloning into 'uftrace'...
remote: Counting objects: 9339, done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 9339 (delta 4), reused 2 (delta 0), pack-reused 9329
Receiving objects: 100% (9339/9339), 2.71 MiB | 178.00 KiB/s, done.
Resolving deltas: 100% (6695/6695), done.
gohkim@ws00837:~/work/tools$ make
make: *** No targets specified and no makefile found. Stop.
gohkim@ws00837:~/work/tools$ cd uftrace/
gohkim@ws00837:~/work/tools/uftrace$ ls
arch         cmd-live.c    cmd-script.c     INSTALL.md        misc       TODO
check-deps   cmd-record.c  configure        libmcount         NEWS       uftrace.c
cmd-dump.c   cmd-recv.c    CONTRIBUTING.md  libtraceevent     README.md  uftrace.h
cmd-graph.c  cmd-replay.c  COPYING          Makefile          scripts    utils
cmd-info.c   cmd-report.c  doc              Makefile.include  tests
gohkim@ws00837:~/work/tools/uftrace$ make
  GEN      .config
  GEN      version.h
  CC       uftrace.o
  CC       cmd-recv.o
  CC       cmd-dump.o
  CC       cmd-report.o
  CC       cmd-replay.o
  CC       cmd-info.o
  CC       cmd-record.o
  CC       cmd-script.o
  CC       cmd-live.o
  CC       cmd-graph.o
  CC       utils/session.o
  CC       utils/rbtree.o
  CC       utils/script.o
  CC       utils/kernel.o
  CC       utils/data-file.o
  CC       utils/pager.o
  CC       utils/demangle.o
  CC       utils/filter.o
  CC       utils/debug.o
  CC       utils/fstack.o
  CC       utils/script-python.o
  CC       utils/symbol.o
  CC       utils/utils.o
  CC       arch/x86_64/cpuinfo.o
  CC       arch/x86_64/regs.o
  LINK     arch/x86_64/uftrace.o
  FLAGS:   * new build flags or cross compiler
  CC FPIC  libtraceevent/event-parse.o
  CC FPIC  libtraceevent/event-plugin.o
  CC FPIC  libtraceevent/trace-seq.o
  CC FPIC  libtraceevent/parse-filter.o
  CC FPIC  libtraceevent/parse-utils.o
  CC FPIC  libtraceevent/kbuffer-parse.o
  LINK     libtraceevent/libtraceevent.a
  LINK     uftrace
  CC FPIC  libmcount/mcount.op
  CC FPIC  libmcount/dynamic.op
  CC FPIC  libmcount/event.op
  CC FPIC  libmcount/record.op
  CC FPIC  libmcount/plthook.op
  CC FPIC  symbol.op
  CC FPIC  debug.op
  CC FPIC  rbtree.op
  CC FPIC  filter.op
  CC FPIC  demangle.op
  CC FPIC  utils.op
  CC FPIC  script.op
  CC FPIC  script-python.op
  ASM      arch/x86_64/mcount.op
  ASM      arch/x86_64/fentry.op
  ASM      arch/x86_64/xray.op
  ASM      arch/x86_64/plthook.op
  CC FPIC  arch/x86_64/mcount-event.op
  CC FPIC  arch/x86_64/mcount-dynamic.op
  CC FPIC  arch/x86_64/mcount-support.op
  CC FPIC  arch/x86_64/regs.op
  LINK     arch/x86_64/mcount-entry.op
  LINK     libmcount/libmcount.so
  CC FPIC  libmcount/mcount-fast.op
  CC FPIC  libmcount/dynamic-fast.op
  CC FPIC  libmcount/event-fast.op
  CC FPIC  libmcount/record-fast.op
  CC FPIC  libmcount/plthook-fast.op
  LINK     libmcount/libmcount-fast.so
  CC FPIC  libmcount/mcount-single.op
  CC FPIC  libmcount/dynamic-single.op
  CC FPIC  libmcount/event-single.op
  CC FPIC  libmcount/record-single.op
  CC FPIC  libmcount/plthook-single.op
  LINK     libmcount/libmcount-single.so
  CC FPIC  libmcount/mcount-fast-single.op
  CC FPIC  libmcount/dynamic-fast-single.op
  CC FPIC  libmcount/event-fast-single.op
  CC FPIC  libmcount/record-fast-single.op
  CC FPIC  libmcount/plthook-fast-single.op
  LINK     libmcount/libmcount-fast-single.so
  CC FPIC  libmcount/mcount-nop.op
  LINK     libmcount/libmcount-nop.so
gohkim@ws00837:~/work/tools/uftrace$ sudo make install
[sudo] password for gohkim: 
  INSTALL  uftrace
  INSTALL  libmcount
  INSTALL  bash-completion
  GEN      uftrace.1
  GEN      uftrace-record.1
  GEN      uftrace-replay.1
  GEN      uftrace-live.1
  GEN      uftrace-report.1
  GEN      uftrace-recv.1
  GEN      uftrace-info.1
  GEN      uftrace-dump.1
  GEN      uftrace-graph.1
  GEN      uftrace-script.1
  INSTALL  man-pages
gohkim@ws00837:~/work/tools/uftrace$ uftrace info
gohkim@ws00837:~/work/tools/uftrace$ strace uftrace info
execve("/usr/local/bin/uftrace", ["uftrace", "info"], [/* 69 vars */]) = 0
brk(NULL)                               = 0x55ebd4ecc000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap(NULL, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f05925d5000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=152330, ...}) = 0
mmap(NULL, 152330, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f05925af000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/x86_64-linux-gnu/libelf.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0000.\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=96824, ...}) = 0
mmap(NULL, 2191744, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f059219b000
mprotect(0x7f05921b2000, 2093056, PROT_NONE) = 0
mmap(0x7f05923b1000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x16000) = 0x7f05923b1000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/librt.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340 \0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=31712, ...}) = 0
mmap(NULL, 2128832, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0591f93000
mprotect(0x7f0591f9a000, 2093056, PROT_NONE) = 0
mmap(0x7f0592199000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6000) = 0x7f0592199000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\200\r\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=14608, ...}) = 0
mmap(NULL, 2109680, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0591d8f000
mprotect(0x7f0591d92000, 2093056, PROT_NONE) = 0
mmap(0x7f0591f91000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f0591f91000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/x86_64-linux-gnu/libstdc++.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\300\270\10\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=1590744, ...}) = 0
mmap(NULL, 3699296, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0591a07000
mprotect(0x7f0591b80000, 2093056, PROT_NONE) = 0
mmap(0x7f0591d7f000, 49152, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x178000) = 0x7f0591d7f000
mmap(0x7f0591d8b000, 12896, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f0591d8b000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\240`\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=142400, ...}) = 0
mmap(NULL, 2217000, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f05917e9000
mprotect(0x7f0591801000, 2097152, PROT_NONE) = 0
mmap(0x7f0591a01000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x18000) = 0x7f0591a01000
mmap(0x7f0591a03000, 13352, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f0591a03000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20\5\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1856752, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f05925ad000
mmap(NULL, 3959200, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0591422000
mprotect(0x7f05915e0000, 2093056, PROT_NONE) = 0
mmap(0x7f05917df000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1bd000) = 0x7f05917df000
mmap(0x7f05917e5000, 14752, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f05917e5000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libz.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\320\37\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=112936, ...}) = 0
mmap(NULL, 2207920, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0591206000
mprotect(0x7f0591221000, 2093056, PROT_NONE) = 0
mmap(0x7f0591420000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1a000) = 0x7f0591420000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libm.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20W\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=1088952, ...}) = 0
mmap(NULL, 3178744, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0590efd000
mprotect(0x7f0591005000, 2093056, PROT_NONE) = 0
mmap(0x7f0591204000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x107000) = 0x7f0591204000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libgcc_s.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\220*\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=92584, ...}) = 0
mmap(NULL, 2188336, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f0590ce6000
mprotect(0x7f0590cfc000, 2093056, PROT_NONE) = 0
mmap(0x7f0590efb000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x15000) = 0x7f0590efb000
close(3)                                = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f05925ab000
mmap(NULL, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f05925a8000
arch_prctl(ARCH_SET_FS, 0x7f05925a8740) = 0
mprotect(0x7f05917df000, 16384, PROT_READ) = 0
mprotect(0x7f0590efb000, 4096, PROT_READ) = 0
mprotect(0x7f0591204000, 4096, PROT_READ) = 0
mprotect(0x7f0591420000, 4096, PROT_READ) = 0
mprotect(0x7f0591a01000, 4096, PROT_READ) = 0
mprotect(0x7f0591d7f000, 40960, PROT_READ) = 0
mprotect(0x7f0591f91000, 4096, PROT_READ) = 0
mprotect(0x7f0592199000, 4096, PROT_READ) = 0
mprotect(0x7f05923b1000, 4096, PROT_READ) = 0
mprotect(0x55ebd4852000, 8192, PROT_READ) = 0
mprotect(0x7f05925d8000, 4096, PROT_READ) = 0
munmap(0x7f05925af000, 152330)          = 0
set_tid_address(0x7f05925a8a10)         = 21007
set_robust_list(0x7f05925a8a20, 24)     = 0
rt_sigaction(SIGRTMIN, {0x7f05917eeb40, [], SA_RESTORER|SA_SIGINFO, 0x7f05917fa670}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {0x7f05917eebd0, [], SA_RESTORER|SA_RESTART|SA_SIGINFO, 0x7f05917fa670}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
getrlimit(RLIMIT_STACK, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
brk(NULL)                               = 0x55ebd4ecc000
brk(0x55ebd4efe000)                     = 0x55ebd4efe000
ioctl(2, TCGETS, {B38400 opost isig icanon echo ...}) = 0
ioctl(1, TCGETS, {B38400 opost isig icanon echo ...}) = 0
rt_sigaction(SIGINT, {0x55ebd4634cb0, [INT], SA_RESTORER|SA_RESTART, 0x7f05914577f0}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGTERM, {0x55ebd4634cb0, [TERM], SA_RESTORER|SA_RESTART, 0x7f05914577f0}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGPIPE, {0x55ebd4634cb0, [PIPE], SA_RESTORER|SA_RESTART, 0x7f05914577f0}, {SIG_DFL, [], 0}, 8) = 0
ioctl(1, TCGETS, {B38400 opost isig icanon echo ...}) = 0
access("/usr/bin/pager", X_OK)          = 0
pipe([3, 4])                            = 0
clone(child_stack=0, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f05925a8a10) = 21008
close(3)                                = 0
dup2(4, 1)                              = 1
ioctl(2, TCGETS, {B38400 opost isig icanon echo ...}) = 0
dup2(4, 2)                              = 2
close(4)                                = 0
open("uftrace.data/info", O_RDONLY)     = -1 ENOENT (No such file or directory)
open("ftrace.dir/info", O_RDONLY)       = -1 ENOENT (No such file or directory)
stat("uftrace.data/info", 0x7ffed2d6b8f0) = -1 ENOENT (No such file or directory)
close(1)                                = 0
close(2)                                = 0
[{WIFEXITED(s) && WEXITSTATUS(s) == 0}], 0, NULL) = 21008
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=21008, si_uid=1001, si_status=0, si_utime=0, si_stime=0} ---
exit_group(-1)                          = ?
+++ exited with 255 +++
```

--- a/cmd-dump.c
+++ b/cmd-dump.c
@@ -1238,8 +1238,10 @@ int command_dump(int argc, char *argv[], struct opts *opts)
 	struct ftrace_file_handle handle;
 
 	ret = open_data_file(opts, &handle);
-	if (ret < 0)
-		pr_err("cannot open data: %s", opts->dirname);
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
+		return -1;
+	}
 
 	fstack_setup_filters(opts, &handle);
 

--- a/cmd-graph.c
+++ b/cmd-graph.c
@@ -651,8 +651,10 @@ int command_graph(int argc, char *argv[], struct opts *opts)
 		func = "main";
 
 	ret = open_data_file(opts, &handle);
-	if (ret < 0)
-		pr_err("cannot open data: %s", opts->dirname);
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
+		return -1;
+	}
 
 	if (opts->depth != OPT_DEPTH_DEFAULT) {
 		/*

--- a/cmd-info.c
+++ b/cmd-info.c
@@ -769,12 +769,17 @@ void clear_ftrace_info(struct uftrace_info *info)
 
 int command_info(int argc, char *argv[], struct opts *opts)
 {
+	int ret;
 	char buf[PATH_MAX];
 	struct stat statbuf;
 	struct ftrace_file_handle handle;
 	const char *fmt = "# %-20s: %s\n";
 
-	open_data_file(opts, &handle);
+	ret = open_data_file(opts, &handle);
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
+		return -1;
+	}
 
 	snprintf(buf, sizeof(buf), "%s/info", opts->dirname);
 

--- a/cmd-info.c
+++ b/cmd-info.c
@@ -783,6 +783,11 @@ int command_info(int argc, char *argv[], struct opts *opts)
 			.flags = SYMTAB_FL_DEMANGLE,
 		};
 
+		if (!opts->exename) {
+			pr_use("Usage: uftrace info --symbols [COMMAND]\n");
+			return -1;
+		}
+
 		load_symtabs(&symtabs, opts->dirname, opts->exename);
 		print_symtabs(&symtabs);
 		unload_symtabs(&symtabs);

--- a/cmd-info.c
+++ b/cmd-info.c
@@ -776,15 +776,6 @@ int command_info(int argc, char *argv[], struct opts *opts)
 	const char *fmt = "# %-20s: %s\n";
 
 	ret = open_data_file(opts, &handle);
-	if (ret < 0) {
-		pr_warn("cannot open data: %s: %m\n", opts->dirname);
-		return -1;
-	}
-
-	snprintf(buf, sizeof(buf), "%s/info", opts->dirname);
-
-	if (stat(buf, &statbuf) < 0)
-		return -1;
 
 	if (opts->print_symtab) {
 		struct symtabs symtabs = {
@@ -797,6 +788,16 @@ int command_info(int argc, char *argv[], struct opts *opts)
 		unload_symtabs(&symtabs);
 		goto out;
 	}
+
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
+		return -1;
+	}
+
+	snprintf(buf, sizeof(buf), "%s/info", opts->dirname);
+
+	if (stat(buf, &statbuf) < 0)
+		return -1;
 
 	pr_out("# system information\n");
 	pr_out("# ==================\n");

--- a/cmd-replay.c
+++ b/cmd-replay.c
@@ -930,8 +930,10 @@ int command_replay(int argc, char *argv[], struct opts *opts)
 	__fsetlocking(logfp, FSETLOCKING_BYCALLER);
 
 	ret = open_data_file(opts, &handle);
-	if (ret < 0)
-		pr_err("cannot open data: %s", opts->dirname);
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
+		return -1;
+	}
 
 	fstack_setup_filters(opts, &handle);
 	setup_field(opts);

--- a/cmd-report.c
+++ b/cmd-report.c
@@ -1129,8 +1129,10 @@ int command_report(int argc, char *argv[], struct opts *opts)
 		avg_mode = AVG_SELF;
 
 	ret = open_data_file(opts, &handle);
-	if (ret < 0)
-		pr_err("cannot open data: %s", opts->dirname);
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
+		return -1;
+	}
 
 	fstack_setup_filters(opts, &handle);
 

--- a/cmd-script.c
+++ b/cmd-script.c
@@ -124,8 +124,10 @@ int command_script(int argc, char *argv[], struct opts *opts)
 	__fsetlocking(logfp, FSETLOCKING_BYCALLER);
 
 	ret = open_data_file(opts, &handle);
-	if (ret < 0)
+	if (ret < 0) {
+		pr_warn("cannot open data: %s: %m\n", opts->dirname);
 		return -1;
+	}
 
 	fstack_setup_filters(opts, &handle);
 

--- a/utils/data-file.c
+++ b/utils/data-file.c
@@ -357,7 +357,7 @@ static void check_data_order(struct ftrace_file_handle *handle)
 }
 
 #define RECORD_MSG  "Was '%s' compiled with -pg or\n"		\
-"\t-finstrument-functions flag and ran with ftrace record?\n"
+"\t-finstrument-functions flag and ran with uftrace record?\n"
 
 int open_data_file(struct opts *opts, struct ftrace_file_handle *handle)
 {
@@ -389,7 +389,7 @@ retry:
 			pr_dbg("cannot find %s file!\n", buf);
 
 			if (opts->exename)
-				pr_err(RECORD_MSG, opts->exename);
+				pr_warn(RECORD_MSG, opts->exename);
 		} else {
 			pr_err("cannot open %s file", buf);
 		}

